src/python/pants/invalidation/build_invalidator.py

```diff
@@ -223,7 +223,7 @@ def _read_sha(self, cache_key):
   def _read_sha_by_id(self, id):
     try:
       with open(self._sha_file_by_id(id), 'rb') as fd:
-        return fd.read().strip()
+        return fd.read().strip().decode('utf-8')
```

Comment:
Preferred to open in mode `'r'`, and then you can avoid the decode. Don't worry about fixing and repushing though - https://github.com/pantsbuild/pants/pull/6372/files#diff-0bfcef172d3faad2ba6f85c7330adae2R225 will fix it.

Thanks again John for your help here!